scripts/s3-sanity/s3-sanity-test.sh

```diff
@@ -17,81 +18,221 @@
+     if [ "$data_path_test" == true ]
+     then
+       ls $s3cmdconfigfile 2> /dev/null || die_with_error "S3cmd configuration file $s3cmdconfigfile is missing"
+       sed -i "s/host_base =.*/host_base = s3.seagate.com/g" $s3cmdconfigfile 
+     fi
+   rm -rf /root/.sgs3iamcli
+}
+
+cleanup() {
+   update_config
+   output=$(s3iamcli resetaccountaccesskey -n SanityAccountToDeleteAfterUse --ldapuser sgiamadmin --ldappasswd "$ldappasswd" --no-ssl) 2> /dev/null || echo ""
+   if [[ $output == *"Name or service not known"*  || $output == *"Invalid argument"* ]]
+   then
+       echo "Provide the Appropriate endpoint"
+       restore_config
+       exit -1
+   elif [[ $output == *"NoSuchEntity"* ]]
+   then
+      echo "Sanity Account doesn't exist for cleanup"
+      return;
+   else
+      access_key=$(echo -e "$output" | tr ',' '\n' | grep "AccessKeyId" | awk '{print $3}')
+      secret_key=$(echo -e "$output" | tr ',' '\n' | grep "SecretKey" | awk '{print $3}')
+      TEST_CMD="s3cmd --access_key=$access_key --secret_key=$secret_key -c $s3cmdconfigfile"
+     if [ "$data_path_test" == true ]
+     then
+       $TEST_CMD del "s3://sanitybucket/SanityObjectToDeleteAfterUse" >/dev/null 2>&1 || echo ""
+       $TEST_CMD rb "s3://sanitybucket" >/dev/null 2>&1 || echo ""
+      fi
+      s3iamcli deleteuser -n SanityUserToDeleteAfterUse --access_key "$access_key" --secret_key "$secret_key" --no-ssl 2> /dev/null || echo ""
+      s3iamcli deleteaccount -n SanityAccountToDeleteAfterUse --access_key "$access_key" --secret_key "$secret_key" --no-ssl 2> /dev/null || echo ""
+      rm -f $test_file_input $test_output_file || echo ""
+   fi
+   restore_config
+}
+
+while getopts ":f:p:e:c:d" o; do
+    case "${o}" in
+        e)
+            end_point=${OPTARG}
+            ;;
+        c)
+            externalcleanup=true
+            ;;
+        d)
+            data_path_test=true
+            ;;
+        p)
+            ldappasswd=${OPTARG}
+            ;;
+        f)
+            s3cmdconfigfile=${OPTARG}
+            ;;
+        *)
+            echo "$USAGE"
+            exit -1
+            ;;
+    esac
+done
+shift $((OPTIND-1))
+
+trap "cleanup" ERR
+
+if [ "$data_path_test" == true ]
+then
+  if command -v s3cmd >/dev/null 2>&1;then
+      printf "\nCheck S3CMD...OK"
+  else
+      die_with_error "\ns3cmd not installed\nPlease install patched version built from <s3server src>/rpms/s3cmd/ or from rpm s3cmd"
+  fi
+  if [ ! -f "$s3cmdconfigfile" ]; then
+      die_with_error "\ns3cmd config file $s3cmdconfigfile not found"
+  fi
+fi
+
+# Check s3iamcli is installed
+if command -v s3iamcli >/dev/null 2>&1; then
+    printf "\nCheck s3iamcli...OK"
+else
+    die_with_error "\ns3iamcli not installed\nPlease install s3iamcli using rpm built from <s3server repo>/rpms/s3iamcli/ or from rpm cortx-s3iamcli"
+fi
+
+if [ -z "$ldappasswd" ]
+then
+  die_with_error "Please provide -p <ldappasswd>"
+fi
+
+systemctl restart s3authserver
```

Comment:
why this is needed ?

Reply:
Can we just check if s3authserver is up and running ?

Reply:
@AjinkyaDhumal  suggested that the check is not needed for the time being.